lib/ansible/modules/windows/win_chocolatey.ps1

```diff
@@ -16,532 +19,543 @@ $params = Parse-Args $args -supports_check_mode $true
-        Fail-Json -obj $result -message "Error upgrading package '$package'"
+    if ($version) {
+        $arguments.Add("--version") > $null
+        $arguments.Add($version) > $null
     }
 
-    if ($verbosity -gt 1) {
-        $result.stdout = $output | Out-String
-    }
-
-    if ("$output" -match ' upgraded (\d+)/\d+ package')
-    {
-        if ($matches[1] -gt 0)
-        {
-            $result.changed = $true
-        }
-    }
-    $result.failed = $false
+    return ,$arguments
 }
 
-
-Function Choco-Install
-{
-    [CmdletBinding()]
-
+Function Install-Chocolatey {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSAvoidUsingUserNameAndPassWordParams", "", Justification="We need to use the plaintext pass in the env vars, also using a SecureString here doesn't make sense considering the source is not secure")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSAvoidUsingPlainTextForPassword", "", Justification="See above")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSAvoidUsingConvertToSecureStringWithPlainText", "", Justification="See above")]
     param(
-        [Parameter(Mandatory=$true)]
-        [string] $package,
-        [string] $version,
-        [bool] $force,
-        [int] $timeout,
-        [bool] $skipscripts,
-        [string] $source,
-        [string] $installargs,
-        [string] $packageparams,
-        [bool] $allowemptychecksums,
-        [bool] $ignorechecksums,
-        [bool] $ignoredependencies,
-        [bool] $allowdowngrade,
-        [bool] $allowprerelease,
-        [string] $proxy_url,
-        [string] $proxy_username,
-        [string] $proxy_password,
-        [string] $architecture
+        [String]$proxy_url,
+        [String]$proxy_username,
+        [String]$proxy_password,
+        [String]$source,
+        [String]$source_username,
+        [String]$source_password
     )
 
-    if (Choco-IsInstalled $package)
-    {
-        if ($state -in ("downgrade", "latest"))
-        {
-            Choco-Upgrade -package $package -version $version -force $force -timeout $timeout `
-                -skipscripts $skipscripts -source $source -installargs $installargs `
-                -packageparams $packageparams -allowemptychecksums $allowemptychecksums `
-                -ignorechecksums $ignorechecksums -ignoredependencies $ignoredependencies `
-                -allowdowngrade $allowdowngrade -proxy_url $proxy_url `
-                -proxy_username $proxy_username -proxy_password $proxy_password `
-                -allowprerelease $allowprerelease -architecture $architecture
-            return
+    $choco_app = Get-Command -Name choco.exe -CommandType Application -ErrorAction SilentlyContinue
+    if ($null -eq $choco_app) {
+        # We need to install chocolatey
+        # Enable TLS1.1/TLS1.2 if they're available but disabled (eg. .NET 4.5)
+        $security_protcols = [Net.ServicePointManager]::SecurityProtocol -bor [Net.SecurityProtocolType]::SystemDefault
```

Comment:
I fixed this in another PR, but you have a typo in `security_protcols`.

Reply:
Thanks, will fix